examples/lang/erc721/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+// Copyright 2018-2019 Parity Technologies (UK) Ltd.
+// This file is part of ink!.
+//
+// ink! is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// ink! is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with ink!.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_core::{
+    env::DefaultSrmlTypes,
+    memory::format,
+    storage,
+    storage::Flush,
+};
+use ink_lang::contract;
+use ink_model::EnvHandler;
+use scale::{Encode, Decode};
+
+pub type Result<T, E> = core::result::Result<T, E>;
+pub type TokenId = u32;
+
+#[derive(Encode, Decode)]
+#[cfg_attr(feature = "ink-generate-abi", derive(Metadata))]
+struct Counter {
+    value: storage::Value<u32>,
+}
```

Comment:
This also requires an `ink_core::storage::alloc::AllocateUsing` implementation.

Reply:
However, I think it isn't necessary to use `storage::Value<u32>` within `Counter` but instead you could simply use `u32` and if you require to store `Counter` in the contract's storage you can still store it as `storage::Value<Counter>`.